flink-connector-aws/flink-connector-dynamodb/src/main/java/org/apache/flink/connector/dynamodb/source/split/DynamoDbStreamsShardSplitSerializer.java

```diff
@@ -0,0 +1,108 @@
+import java.io.ByteArrayOutputStream;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.time.Instant;
+
+/**
+ * Serializes and deserializes the {@link DynamoDbStreamsShardSplit}. This class needs to handle
+ * deserializing splits from older versions.
+ */
+@Internal
+public class DynamoDbStreamsShardSplitSerializer
+        implements SimpleVersionedSerializer<DynamoDbStreamsShardSplit> {
+
+    private static final int CURRENT_VERSION = 0;
+
+    @Override
+    public int getVersion() {
+        return CURRENT_VERSION;
+    }
+
+    @Override
+    public byte[] serialize(DynamoDbStreamsShardSplit split) throws IOException {
+        try (ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                DataOutputStream out = new DataOutputStream(baos)) {
+
+            out.writeUTF(split.getStreamArn());
+            out.writeUTF(split.getShardId());
+            out.writeUTF(split.getStartingPosition().getShardIteratorType().toString());
+            if (split.getStartingPosition().getStartingMarker() == null) {
+                out.writeBoolean(false);
+            } else {
+                out.writeBoolean(true);
+                Object startingMarker = split.getStartingPosition().getStartingMarker();
+                out.writeBoolean(startingMarker instanceof Instant);
+                if (startingMarker instanceof Instant) {
+                    out.writeLong(((Instant) startingMarker).toEpochMilli());
+                }
+                out.writeBoolean(startingMarker instanceof String);
+                if (startingMarker instanceof String) {
+                    out.writeUTF((String) startingMarker);
+                }
+            }
+            out.flush();
+            return baos.toByteArray();
+        }
+    }
+
+    @Override
+    public DynamoDbStreamsShardSplit deserialize(int version, byte[] serialized)
+            throws IOException {
+        try (ByteArrayInputStream bais = new ByteArrayInputStream(serialized);
+                DataInputStream in = new DataInputStream(bais)) {
+            if (version != getVersion()) {
+                throw new VersionMismatchException(
+                        "Trying to deserialize DynamoDbStreamsShardSplit serialized with unsupported version "
+                                + version
+                                + ". Version of serializer is "
+                                + getVersion());
+            }
+
+            final String streamArn = in.readUTF();
+            final String shardId = in.readUTF();
+            final ShardIteratorType shardIteratorType = ShardIteratorType.fromValue(in.readUTF());
+            Object startingMarker = null;
+
+            final boolean hasStartingMarker = in.readBoolean();
+            if (hasStartingMarker) {
+                if (in.readBoolean()) {
+                    startingMarker = Instant.ofEpochMilli(in.readLong());
+                }
```

Comment:
This is not tested atm!
<img width="1226" alt="image" src="https://github.com/apache/flink-connector-aws/assets/35062175/f931d573-79df-4103-a4f2-e363c410b5a5">

Reply:
Removed this condition. Marker cannot be a timestamp for DDB Streams